src/Microsoft.AspNetCore.Dispatcher/MatcherLoggerExtensions.cs

```diff
@@ -0,0 +1,190 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Dispatcher
+{
+    internal static class MatcherLoggerExtensions
+    {
+        // MatcherBase and HttpMethodEndpointSelector
```

Comment:
We should not reuse these things. Each class does its own logging and that's independent